tests/microos/install_image.pm

```diff
@@ -0,0 +1,100 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Install SL Micro image on bare metal disk
+# Maintainer: Petr Cervinka <pcervinka@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+use Utils::Backends;
+use Utils::Architectures 'is_x86_64';
+use power_action_utils 'power_action';
+use serial_terminal 'select_serial_terminal';
+
+sub run {
+    select_serial_terminal;
+
+    # Use image name from HDD_1 variable
+    my $image = get_required_var('HDD_1');
+    # Use sda as target disk by default
+    my $device = get_var("MICRO_INSTALL_IMAGE_TARGET_DEVICE", "/dev/sda");
+    # Use partition prefix for nvme devices
+    my $prefix = $device =~ /nvme/ ? "p" : "";
+    # SL Micro x86_64 image has three partitions, aarch64 and ppc64le images have only two partitions
+    my $root_partition_id = is_x86_64 ? 3 : 2;
+    my $root_partition = "${device}" . $prefix . $root_partition_id;
+    # New partition id for ignition will be directly after root partition
+    my $ignition_partition_id = $root_partition_id + 1;
+    my $ignition_partition = "${device}" . $prefix . $ignition_partition_id;
+    record_info("Device information", "Device: ${device}\nRoot partition: ${root_partition}\nIgnition partition: ${ignition_partition}");
+
+    # Mount nfs share with images
+    assert_script_run("mount -o ro,noauto,nofail,nolock -t nfs openqa.suse.de:/var/lib/openqa/share /mnt");
+    assert_script_run("ls -als /mnt/factory/hdd/${image}");
+    # dd image to disk
+    assert_script_run("xzcat /mnt/factory/hdd/${image} | dd of=${device} bs=65536 status=progress", timeout => 300);
+    assert_script_run("sync");
+    assert_script_run("umount /mnt");
+
+    my $device_layout = script_output("lsblk");
+    record_info("Device layout", ${device_layout});
+
+    # Modify disk to be able to correctly boot and login
+    assert_script_run("mount ${root_partition} /mnt");
+    assert_script_run("btrfs property set /mnt ro false");
+    # Set correct serial console to be able to see login in first boot
+    assert_script_run("sed -i 's/console=ttyS0,115200/console=ttyS1,115200/g' /mnt/boot/grub2/grub.cfg") if is_x86_64;
+    # Upload original grub configuration
+    upload_logs("/mnt/etc/default/grub", failok => 1);
+    # Set permanent grub configuration
+    assert_script_run("sed -i 's/console=ttyS0,115200/console=ttyS1,115200/g' /mnt/etc/default/grub") if is_x86_64;
+    # Fully disable graphical terminal on legacy systems without UEFI
```

Comment:
We ever confirmed with SLM project managers that SLE Micro only supports uefi.

Reply:
It doesn't matter at all,  especially for kernel testing where we have more legacy machines than in uefi mode.

Reply:
Really?
How many machines relevant today cannot do EFI?

Reply:
> It doesn't matter at all, especially for kernel testing where we have more legacy machines than in uefi mode.

I do not quite agree . It will brings invalid test results, I think. @jlausuch  HDYT?

Reply:
The ability to be able to install SL Micro on hardware with legacy boot doesn't harm or breaking anything and it's up to particular team where they will schedule tests.

Reply:
yeah this is specific part is in the newly introduced .pm module, so personally I see no reason to block the merge.
We could improve handling of this parameter and we could make it in a way that the test run could change the behavior, but this could be done later when it is needed and by folks who needed it.

Reply:
> Even if it's not the case it would be much more reasonable get the console settings from the commandline of the runnig kernel rather than hardcoding it.

Hardcoded console settings are directly in image produced by ibs. If you think  it is not right, feel free to file a bug or create jira for it.

Reply:
These are, however, additional hardcoded setting in this test, separate from settings hardcoded in the image.

The value with which the settings are replaced could be taken from the commandline of the running kernel which can in turn be adjusted in the test setup making it easier to adjust console settings for the tests.

Reply:
Could be, but  console of running kernel is already configured  by the test in ipxe boot script from variable IPXE_CONSOLE.  Follow up ticket for this change was already created and is mentioned on other comments in this PR.

Reply:
https://progress.opensuse.org/issues/167608